alvr/common/src/lib.rs

```diff
@@ -1,19 +1,34 @@
 pub mod logging;
-pub mod version;
 
-#[cfg(not(target_os = "android"))]
-pub mod commands;
-
-pub use logging::*;
-pub use version::*;
+use lazy_static::lazy_static;
+use semver::{BuildMetadata, Prerelease, Version};
 
 pub type StrResult<T = ()> = Result<T, String>;
 
+pub const ALVR_NAME: &str = "ALVR";
+
 pub mod prelude {
     pub use crate::{
-        fmt_e,
-        logging::{log_event, Event},
-        trace_err, trace_err_dbg, trace_none, trace_str, StrResult,
+        fmt_e, logging::*, trace_err, trace_err_dbg, trace_none, trace_str, StrResult,
     };
     pub use log::{debug, error, info, warn};
 }
+
+lazy_static! {
+    pub static ref ALVR_VERSION: Version = Version::parse(env!("CARGO_PKG_VERSION")).unwrap();
+}
+
+// accept semver-compatible versions
+// Note: by not having to set the requirement manually, the major version is constrained to be
+// bumped when the packet layouts or some critical behaviour has changed.
+pub fn is_version_compatible(other_version: &Version) -> bool {
+    if other_version.pre != Prerelease::EMPTY
+        || other_version.build != BuildMetadata::EMPTY
+        || ALVR_VERSION.pre != Prerelease::EMPTY
+        || ALVR_VERSION.build != BuildMetadata::EMPTY
+    {
+        *other_version == *ALVR_VERSION
+    } else {
+        other_version.major == ALVR_VERSION.major
+    }
+}
```

Comment:
Can we have changes like this in a separate PR in the future?

Reply:
As I said on Discord, I just appended more commits after I was done with the trickier part of this PR. If this PR was merged before I would have pushed these commits directly to master branch.